Core/GDCore/Project/Variable.cpp

```diff
@@ -189,23 +235,35 @@ void Variable::RemoveRecursively(const gd::Variable& variableToRemove) {
       it++;
     }
   }
-  isStructure = !children.empty();
+  for (auto it = childrenList.begin(); it != childrenList.end();) {
+    if (it->get() == &variableToRemove) {
+      childrenList.erase(it);
+    } else {
+      it->get()->RemoveRecursively(variableToRemove);

```

Comment:
```suggestion
      it->RemoveRecursively(variableToRemove);
```

I think (or `(*it)->`, but no need for get)